src/posts/entities/postRepository.ts

```diff
@@ -0,0 +1,18 @@
+import { EntityRepository, Repository } from 'typeorm';
+import { CreatePostDto } from '../dto/create-post.dto';
+import { Post } from './post.entity';
+
+@EntityRepository(Post)
+export class PostRepository extends Repository<Post> {
+  public async createPost(createPostDto: CreatePostDto): Promise<Post> {
+    const { caption, is_hidden, user_id, created } = createPostDto;
+    const post = new Post();
```

Comment:
Since we use data mapping paradigm, remove `const post = new Post();` and add `const post = this.create();` instead

Reply:
as stated in the typeorm docs both are the same,  const post = new Post() == const post = this.create(); .

Reply:
yes, all these methods will give the same result. However, those are different paradigms, data mapper vs active record. I understood this from docs also